sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -39,29 +42,362 @@ class ConfigurationSetting(Model):
+                    key_value
+                )
+            if key_value.content_type.startswith(
+                SecretReferenceConfigurationSetting._secret_reference_content_type  # pylint:disable=protected-access
+            ):
+                return SecretReferenceConfigurationSetting._from_generated(  # pylint: disable=protected-access
+                    key_value
+                )
+
         return cls(
             key=key_value.key,
             label=key_value.label,
-            content_type=key_value.content_type,
             value=key_value.value,
+            content_type=key_value.content_type,
             last_modified=key_value.last_modified,
             tags=key_value.tags,
             read_only=key_value.locked,
             etag=key_value.etag,
         )
+
+    def _to_generated(self):
+        # type: (...) -> KeyValue
+        return KeyValue(
+            key=self.key,
+            label=self.label,
+            value=self.value,
+            content_type=self.content_type,
+            last_modified=self.last_modified,
+            tags=self.tags,
+            locked=self.read_only,
+            etag=self.etag,
+        )
+
+
+class FeatureFlagConfigurationSetting(
+    ConfigurationSetting
+):  # pylint: disable=too-many-instance-attributes
+    """A feature flag configuration value.
+    Variables are only populated by the server, and will be ignored when
+    sending a request.
+    :ivar etag: Entity tag (etag) of the object
+    :vartype etag: str
+    :ivar key:
+    :vartype key: str
+    :ivar enabled:
+    :vartype enabled: bool
+    :param filters:
+    :type filters: list[dict[str, Any]]
+    :param label:
+    :type label: str
+    :param content_type:
+    :type content_type: str
+    :ivar last_modified:
+    :vartype last_modified: datetime
+    :ivar read_only:
+    :vartype read_only: bool
+    :param tags:
+    :type tags: dict[str, str]
+    """
+
+    _attribute_map = {
+        "etag": {"key": "etag", "type": "str"},
+        "key": {"key": "key", "type": "str"},
+        "label": {"key": "label", "type": "str"},
+        "content_type": {"key": "content_type", "type": "str"},
+        "value": {"key": "value", "type": "str"},
+        "last_modified": {"key": "last_modified", "type": "iso-8601"},
+        "read_only": {"key": "read_only", "type": "bool"},
+        "tags": {"key": "tags", "type": "{str}"},
+    }
+    key_prefix = ".appconfig.featureflag/"
+    _feature_flag_content_type = (
+        "application/vnd.microsoft.appconfig.ff+json;charset=utf-8"
+    )
+    kind = "FeatureFlag"
+
+    def __init__(self, key, enabled, filters=None, **kwargs):
+        # type: (str, bool, Optional[List[Dict[str, Any]]]) -> None
+        self._enabled = enabled
+        super(FeatureFlagConfigurationSetting, self).__init__(**kwargs)
+        self.key = key
+        if not key.startswith(self.key_prefix):
+            self.key = self.key_prefix + key
```

Comment:
This is interesting. What's the behavior of other languages?

Reply:
This only happens when a user creates one in a Python program. On deserialization we only deserialize into `FeatureFlagConfigurationSetting` if the prefix and content_type are the correct value.

Reply:
I mean if user creates a FeatureFlagConfigurationSetting with key ".appconfig.featureflag/abc"

How can we know he/she wants ".appconfig.featureflag/abc" or ".appconfig.featureflag/.appconfig.featureflag/abc"?

Reply:
On the other hand do we want a user to have to create a feature flag by doing:
```python
my_flag = FeatureFlagConfigurationSetting(".appconfig.featureflag/my_flag", enabled=True)
```
or
```python
my_flag = FeatureFlagConfigurationSetting("my_flag", enabled=True)
```
@xiangyan99

Reply:
I don't have a strong opinion on this. Let's align the behavior with other languages. :)

Reply:
Talked with other languages. I will make the key prefix a constant and export it.